src/main/java/org/squiddev/plethora/integration/hatchery/IntegrationHatchery.java

```diff
@@ -0,0 +1,287 @@
+
+			//Somewhat of a misnomer; this is the 'energy' level, comparable to a furnace's remaining burn ticks
+			//The actual 'temptation items' (Hatchery's term, not mine) is exposed via the Item Handler Capability,
+			//TODO How can I expose the Rooster's inventory in a manner consistent with the rest of the code?
+			// I can't bind an Introspection Module to it...
+			out.put("seeds", rooster.getSeeds());
+
+			return out;
+		}
+
+		@Nullable
+		@Override
+		public EntityRooster getExample() {
+			//REFINE See META_HATCHERY_EGG#getExample's task
+			ResourceLocation roosterLocation = EntityList.getKey(EntityRooster.class);
+			if (roosterLocation == null) return null;
+
+			Entity rooster = EntityList.createEntityByIDFromName(roosterLocation, WorldDummy.INSTANCE);
+			if (!(rooster instanceof EntityRooster)) return null;
+
+			return (EntityRooster) rooster;
+		}
+	};
+
+	public static final IMetaProvider<NestPenTileEntity> META_NESTPEN = new BaseMetaProvider<NestPenTileEntity>() {
+
+		@Nonnull
+		@Override
+		public Map<Object, Object> getMeta(@Nonnull IPartialContext<NestPenTileEntity> context) {
+			Map<Object, Object> out = new HashMap<>();
+			NestPenTileEntity target = context.getTarget();
+
+			//Get the stored entity
+			EntityAgeable storedEntity = target.storedEntity();
+			if (storedEntity != null) out.put("storedEntity", context.makePartialChild(storedEntity).getMeta());
+
+			//The stored entity and the inventory are straightforward
+			//The only other data that we _may_ want to expose would be the time to next drop
+
+			return out;
+
+		}
+
+		@Nonnull
+		@Override
+		public NestPenTileEntity getExample() {
+			NestPenTileEntity te = new NestPenTileEntity();
+			Entity chicken = EntityList.createEntityByIDFromName(new ResourceLocation("minecraft:chicken"), WorldDummy.INSTANCE);
+
+			te.trySetEntity(chicken);
+			return te;
+		}
+	};
+
+	public static final IMetaProvider<EggNestTileEntity> META_EGG_NEST = new BaseMetaProvider<EggNestTileEntity>() {
+
+		@Nonnull
+		@Override
+		public Map<Object, Object> getMeta(@Nonnull IPartialContext<EggNestTileEntity> context) {
+			Map<Object, Object> out = new HashMap<>();
+			EggNestTileEntity target = context.getTarget();
+
+			//The 'hasEgg' property is exposed as part of the BlockState
+
+			ItemStack egg = target.getEgg();
+			if (egg != null) {
+				out.put("egg", context.makePartialChild(egg).getMeta());
+				out.put("hatchPercent", target.getPercentage());
+			}
+
+			//The nest also obtains hatching bonuses from nearby entities, (especially players?),
+			// as well as from 'heat lamps' (powered RS lamps above the nest).
+			// However, these checks are in private code, and I'd rather not replicate something
+			// that may be trivially changed in an update.
+
+			return out;
+
+		}
+
+		@Nonnull
+		@Override
+		public EggNestTileEntity getExample() {
+			EggNestTileEntity te = new EggNestTileEntity();
+			ItemStack egg = new ItemStack(Items.EGG);
+
+			te.insertEgg(egg);
```

Comment:
Really petty (sorry), but remove the intermediate variable.

Reply:
Done, pending commit